sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -102,37 +111,48 @@ void ng_ipv6_netif_init(void)
 
 void ng_ipv6_netif_add(kernel_pid_t pid)
 {
+    ng_ipv6_netif_t *free_entry = NULL;
+
     for (int i = 0; i < NG_NETIF_NUMOF; i++) {
         if (ipv6_ifs[i].pid == pid) {
-            return; /* prevent duplicates */
+            /* pid has already been added */
+            return;
+        }
+
+        else if ((ipv6_ifs[i].pid == KERNEL_PID_UNDEF) && !free_entry) {
+            /* found the first free entry */
+            free_entry = &ipv6_ifs[i];
         }
-        else if (ipv6_ifs[i].pid == KERNEL_PID_UNDEF) {
-            ng_ipv6_addr_t addr = NG_IPV6_ADDR_ALL_NODES_LINK_LOCAL;
-            mutex_lock(&ipv6_ifs[i].mutex);
+    }
+
+    if (!free_entry) {
```

Comment:
114 needs to be initialized to `NULL`, otherwise this check might succeed, leading to a SegFault below, because `free_entry` would reference some crap previously allocated at this position in the stack.